src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -296,6 +367,14 @@ private GeneratorDriver BuildFinalCompilation(Compilation compilation, out Compi
             return FromState(state);
         }
 
+        private GeneratorState SetGeneratorException(GeneratorState generatorState, ISourceGenerator generator, Exception e, DiagnosticBag diagnosticBag, bool isInit = false)
```

Comment:
>private [](start = 8, length = 7)

`static`

Reply:
It uses `MessageProvider` which is instance. Better to pass it in + make static?

---
In reply to: [466555520](https://github.com/dotnet/roslyn/pull/46296#discussion_r466555520) [](ancestors = 466555520)

Reply:
> It uses MessageProvider which is instance. Better to pass it in + make static?

I tend to prefer `static` functions if there's just one field needed, but either approach seems reasonable.

---
In reply to: [466612020](https://github.com/dotnet/roslyn/pull/46296#discussion_r466612020) [](ancestors = 466612020,466555520)